src/server/apis/check-pr-state.ts

```diff
@@ -0,0 +1,95 @@
+import * as express from 'express';
+import gql from 'graphql-tag';
+
+import {CheckPRPayload, NotificationPullRequestData} from '../../types/api';
+import {PRStateQuery} from '../../types/gql-types';
+import {github} from '../../utils/github';
+// import {github} from '../../utils/github';
+import {userModel} from '../models/userModel';
+
+function getRouter(): express.Router {
+  const checkPRStateRouter = express.Router();
+  checkPRStateRouter.post(
+      '/', async (request: express.Request, response: express.Response) => {
+        try {
+          if (!request.body) {
+            response.status(400).send('No body.');
+            return;
+          }
+
+          const loginDetails = await userModel.getLoginFromRequest(request);
+          if (!loginDetails) {
+            response.status(400).send('No login details.');
+            return;
+          }
+
+          const pullRequests = request.body;
+
+          const prIds: string[] = [];
+          for (const pullRequest of pullRequests) {
+            if (!pullRequest.gqlId) {
```

Comment:
As far as I can tell this is incorrect. It seems that the request is just an array of strings. ie `['id1', 'id2']`. If that's true, then this entire loop isn't necessary: `prIds = request.body`.